src/etl/ReportingETL.cpp

```diff
@@ -208,6 +217,20 @@ ReportingETL::publishLedger(ripple::LedgerInfo const& lgrInfo)
     else
         BOOST_LOG_TRIVIAL(info) << __func__ << " - Skipping publishing ledger "
                                 << std::to_string(lgrInfo.seq);
+    if (!writing_)
+    {
+        BOOST_LOG_TRIVIAL(debug) << __func__ << " - Updating cache";
+
+        std::vector<Backend::LedgerObject> diff =
+            Backend::synchronousAndRetryOnTimeout([&](auto yield) {
+                return backend_->fetchLedgerDiff(lgrInfo.seq, yield);
+            });
+
+        backend_->cache().update(diff, lgrInfo.seq);
+        backend_->txCache().update(hashes, transactions, lgrInfo.seq);
```

Comment:
If we're updating the `txCache_` here, we don't need to also update it in `insertTransactions`. 

`publishLedger` is called in both readonly and etl modes.

Reply:
The condition to enter this block is `if (!writing_)`

The intention is that for writing nodes the cache is updated in insertTransactions, while for reading nodes it is updated in publish ledger.

Reply:
So then does `backend_->cache() not get updated in etl mode?

Reply:
@cjcobb23 I think that will slow down the extraction of successors a lot, right?